aten/src/ATen/function_wrapper.py

```diff
@@ -596,14 +609,22 @@ def process_native(option, output_options):
                             "but specified for function {}", option['name'])
 
         def_level = option.get('type_method_definition_level')
-        if def_level != 'base':
-            raise Exception("\'base\' is currently the only supported type_method_definition_level "
-                            "for native functions, got level {} for {}", def_level, option['name'])
+        if def_level != 'base' and def_level != 'backend':
```

Comment:
In a comment somewhere,  there should be an explanation what `type_method_definition_level` is. My vote is for the top of the native functions implementation file.

Reply:
Will do.

Reply:
This logic feels more complicated than it needs to be, particularly because there seems like there are too many degrees of freedom by having both type_method_definition_level and type_method_definition_dispatch.

Maybe just use one field `dispatch`: 

Dispatch happens at Type (current base)
```
- dispatch: at::native::myfunction
```

Dispatch happens per backend:
```
dispatch:
     CPU: at::native::my_func
     CUDA: at::native::my_func_cuda
```
This would avoid needing to explain type_method_definition_dispatch since we would no longer have it.

Reply:
agreed.